stock_warehouse_orderpoint_stock_info_unreserved/models/stock_warehouse_orderpoint.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class StockWarehouseOrderpoint(models.Model):
+    _inherit = 'stock.warehouse.orderpoint'
+
+    @api.multi
+    def _compute_product_available_qty(self):
+        super(StockWarehouseOrderpoint, self)._compute_product_available_qty()
```

Comment:
Dear I have made a PR to compute fast https://github.com/OCA/stock-logistics-warehouse/pull/340/

Please find a code to replace, like that the methode `_product_available()` (replaced by `_compute_quantities_dict()` in new version), is computed once by location.

```
op_by_loc = defaultdict(lambda: self.env['stock.warehouse.orderpoint'])
        for order in self:
            op_by_loc[order.location_id] |= order
        for location_id, order_in_loc in op_by_loc.items():
            products = order_in_loc.mapped('product_id').with_context(
                location=location_id.id)._compute_quantities_dict(
                lot_id=None, owner_id=None, package_id=None)
            for order in order_in_loc:
                product = products[order.product_id.id]
                order.product_location_qty_available_not_res = products['qty_available_not_res']
```

cc @sylvain-garancher

Reply:
Done